packages/just-task/src/cache.ts

```diff
@@ -0,0 +1,72 @@
+import { getPackageDeps } from '@microsoft/package-deps-hash';
+import { argv } from './option';
+import { resolveCwd } from './resolve';
+import fs from 'fs-extra';
+import path from 'path';
+import { logger } from 'just-task-logger';
+
+const cachedTask: string[] = [];
+
+export function registerCachedTask(taskName: string) {
+  cachedTask.push(taskName);
+}
+
+export function isCached(taskName: string) {
+  if (cachedTask.indexOf(taskName) < 0) {
+    return false;
+  }
+
+  const currentHash = getHash(taskName);
+  const cachePath = getCachePath();
+  const cacheFile = path.join(cachePath, 'package-deps.json');
+
+  if (!fs.existsSync(cacheFile)) {
+    return false;
+  }
+
+  if (!fs.pathExistsSync(cachePath)) {
+    fs.mkdirpSync(cachePath);
+  }
+
+  try {
+    const cachedHash = JSON.parse(fs.readFileSync(path.join(cachePath, 'package-deps.json')).toString());
```

Comment:
Does this mean only the last cached task wins? For example if I did `npm run build`, `npm run ut`, `npm run build`, would the last build rebuild since the ut task would overwrite the contents of the cache?

Reply:
Yeah, if you're doing npm run build, then run ut, the cache would be what was written for UT. I can't think of a good way to tackle this kind of thing without something like the docker image layers. That required a bunch more infra. Maybe sometime in the future I can look to other tech than node + @microsoft/package-deps-hash... but this gives some quick wins when you're just rebuilding a bunch of packages in a monorepo where you just change one package.

Reply:
What you described is the right behavior I think. That's what I have.

Reply:
Is it possible to make package-deps.json keyed by script name? Like have 

```json
{
  "ut": [hash info]
   "build": [hash info]
   "webpack": [hash info] 
}
```

I imagine there is complexity around this. For example running ut's shouldn't invalidate the build cache, but running build may invalidate the ut cache (depending on the project) and will very likely invalidate the webpack hash (since it likely depends on build). Maybe we should start with simple?

But then again, I wonder how we'd enable build caching at the CI level without this script level caching capability